Samples/WindowPlacement/README.md

```diff
@@ -0,0 +1,858 @@
+ - [SetWindowPos](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowpos)
+ - [GetMonitorInfo](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getmonitorinfoa)
+ - [GetWindowPlacement](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getwindowplacement)
+ - [SetWindowPlacement](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowplacement)
+ - [GetDpiForWindow](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getdpiforwindow)
+ - [IsWindowArranged](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-iswindowarranged)
+ - [GetDpiForMonitor](https://learn.microsoft.com/en-us/windows/win32/api/shellscalingapi/nf-shellscalingapi-getdpiformonitor)
+ - [DwmGetWindowAttribute](https://learn.microsoft.com/en-us/windows/win32/api/dwmapi/nf-dwmapi-dwmgetwindowattribute)
+ - [IVirtualDesktopManager](https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-ivirtualdesktopmanager)
+
+## Overview
+
+Most users expect apps to launch where they last closed. For example, apps
+closed on secondary monitors should relaunch on that monitor (and not on the
+primary). Or if maximized, apps should relaunch maximized (not at their restore
+position).
+
+Most apps today attempt to do this. The
+[`SetWindowPlacement`](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowplacement)
+and
+[`GetWindowPlacement`](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getwindowplacement)
+APIs assist with this. But even when using these APIs correctly, sometimes apps
+launch in unexpected places, especially for users with monitors that have
+different DPI scales, or whose machines reboot or dock/undock frequently.
+
+This happened because restore was initially (~Win3.1) fairly simple and grew in
+complexity over many decades. **Multiple monitors**, **DPI**, **Arrangement**,
+and **Virtual Desktops** in particular make positioning top-level windows more
+complicated, which in turn makes storing window positions (and using them later)
+more complicated. This README, and [PlacementEx.h](PlacementEx.h), introduce
+fresh guidance for storing window positions.
+
+## Background
+
+### Screen Coordinates
+
+Windows (HWNDs) parented to the Desktop window are called [**top-level windows**](https://learn.microsoft.com/en-us/windows/win32/winmsg/about-windows#parent-or-owner-window-handle).
+These windows can move freely between all monitors. (As opposed to **child windows**,
+which move relative to their parent).
+
+The coordinate space of the Desktop window, the one top-level windows move within,
+is called **Screen Coordinates**.
+
+There is always a monitor connected to the system whose top-left corner is 0,0
+in screen coordinates. This is the primary monitor.
+
+Monitors to the left of the primary have negative X values, and ones above the
```

Comment:
Clarify even on RTL. Per offline feedback:

> You can have a mix of windows on the screen, some LTR and some RTL. So it doesn't make sense to say "the screen is RTL" because the screen could be a mix of some LTR and some RTL.
>
> Therefore, the screen is always LTR. It's the only sane choice.
>
> That's why the docs say not to use ScreenToClient in RTL scenarios - it doesn't do what you want.